src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SHAHashProvider.Browser.Native.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Diagnostics;
+using System.Security.Cryptography;
+
+using SimpleDigest = Interop.BrowserCrypto.SimpleDigest;
+
+namespace Internal.Cryptography
+{
+    internal sealed class SHANativeHashProvider : HashProvider
+    {
+        private readonly int hashSizeInBytes;
+        private readonly SimpleDigest impl;
+        private MemoryStream? buffer;
+
+        public SHANativeHashProvider(string hashAlgorithmId)
+        {
+            Debug.Assert(HashProviderDispenser.CanUseSubtleCryptoImpl);
+
+            switch (hashAlgorithmId)
+            {
+                case HashAlgorithmNames.SHA1:
+                    impl = SimpleDigest.Sha1;
+                    hashSizeInBytes = 20;
+                    break;
+                case HashAlgorithmNames.SHA256:
+                    impl = SimpleDigest.Sha256;
+                    hashSizeInBytes = 32;
+                    break;
+                case HashAlgorithmNames.SHA384:
+                    impl = SimpleDigest.Sha384;
+                    hashSizeInBytes = 48;
+                    break;
+                case HashAlgorithmNames.SHA512:
+                    impl = SimpleDigest.Sha512;
+                    hashSizeInBytes = 64;
+                    break;
+                default:
+                    throw new CryptographicException(SR.Format(SR.Cryptography_UnknownHashAlgorithm, hashAlgorithmId));
+            }
+        }
+
+        public override void AppendHashData(ReadOnlySpan<byte> data)
+        {
+            buffer ??= new MemoryStream(1000);
```

Comment:
What's the value in MemoryStream over `List<byte>`?

Reply:
Aaron was following what was already in the implementation - https://github.com/dotnet/runtime/blob/b423166a1756aec2a8835e7e2c84091ff2fd66fc/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/SHAHashProvider.Browser.cs#L14-L16 I'm unsure of the capacity characteristics of the input to this method, so I Ieft as-is.